apps/endatix-hub/app/(main)/forms/ui/create-form-sheet.tsx

```diff
@@ -80,10 +81,11 @@ const CreateFormSheet = () => {
             formDefinitionJsonData: JSON.stringify("{ }")
         }
         const formResult = await createFormAction(request);
-        if (formResult.isSuccess && formResult.formId) {
-            redirect(`/forms/${formResult.formId}`);
+        if (Result.isSuccess(formResult) && formResult.value) {
+            const formId = formResult.value;
+            redirect(`/forms/${formId}`);
         } else {
-            alert(formResult.error);
```

Comment:
I see what the problem is. Error only has message and errorType as required props. If you don't see them in the auto-complete, then let me know. 

```type Error = {
    kind: Kind.Error;
    errorType: ErrorType;
    message: string;
    details?: string;
};
```

For the future, I will change the API to return decomposed object with the possible states, e.g.:

```ts
const [success, error] = createFormResult();
if (error){
//return early with error
}

console.log(success.value); // happy path
```

Reply:
I couldn't see something about error. Now I think maybe it should be message but strangely it doesn't appear in the auto-complete.